Documentation/Installation/EnvironmentConfiguration.rst

```diff
@@ -2,35 +2,197 @@
+    :ref:`t3coreapi:sitehandling-using-env-vars`.
+
+The following sections describe this implementation process in depth.
+
+.. _environment-dotenv:
+
+.env / dotenv files
+===================
+
+A central advantage of :file:`.env` files is that environment variables can
+also be set in CLI context or injected via Continuous Integration/Deploment
+(CI/CD) systems (GitLab/GitHub) or even Webserver configuration. A central place
+for environment-specific configuration, which many other applications refer to
+(and is easy to locate) is also a best practice.
+
+To let your TYPO3 configuration parse keys and values stored of such a file,
+you need a library like https://github.com/symfony/dotenv/ or
+https://github.com/vlucas/phpdotenv/ and parse it in your :file:`system/additional.php`
+
+You can require these libraries through Packagist/Composer or by including the dependency
+in your project manually (legacy installation).
+
+Example for `symfony/dotenv`:
+
+..  literalinclude:: Environments/_dotenv-symfony.php
+    :language: php
+    :caption: <config>/system/additional.php
+
+Example for `vlucas/phpdotenv`:
+
+..  literalinclude:: Environments/_dotenv-vlucas.php
+    :language: php
+    :caption: <config>/system/additional.php
+
+Once this code has loaded the :php:`$_ENV` variables from the specified file,
+you can access contents of the variables anywhere you need.
+
+.. _environment-helhum-dotenv:
+
+helhum/dotenv-connect
+---------------------
+
+You can also use https://github.com/helhum/dotenv-connector/ (via the Packagist
+package `helhum/dotenv-connector`) to allow accessing :php:`$_ENV` variables
+directly within the Composer autoload process.
+
+This has two nice benefits:
+
+*   You can even set the `TYPO3_CONTEXT` application context environment variable
+    through an :file:`.env` file, and no longer need to specify that in your webserver
+    configuration (e.g. via `.htaccess` or VirtualHost-Configuration).
+*   You do not need to add and maintain such loading code to your :file:`additional.php`
+    file.
+
+The drawback is that you'll have an additional dependency on another package, and
+glue code that is outside of your own implementation.
+
+.. _environment-phpconfig:
+
+Plain PHP configuration files
+=============================
+
+If the concept of requiring a specific file format and their loader dependencies
+seems like too much overhead for you, something similar can be achieved
+by including environment-specific :php:`.php` files
+
+For example, you can create a custom file like :file:`system/environment.php` that
+will only be placed on your specific target server (and not be kept in your versioning
+control system).
+
+..  literalinclude:: Environments/_environment.php
+    :language: php
+    :caption: <config>/system/environment.php
+
+This file would also need to be loaded through the additional configuration
+workflow:
+
+..  literalinclude:: Environments/_additional-native.php
```

Comment:
The content is the same is in _environment.php, but should load the _environment.php AFAIU.

Reply:
Thanks, clarified. I made a boo-boo there when changing files, because additional-native.php is versioned and loads environment.php (I thought that was a bug when I reviewd this)